seqr/utils/elasticsearch/es_search.py

```diff
@@ -1020,10 +1022,10 @@ def _quality_filters_by_family(quality_filter, samples_by_family_index, indices)
     if any(quality_filter[field] for field in quality_field_configs.keys()):
         for index in indices:
             family_samples_by_id = samples_by_family_index[index]
-            for family_guid, samples_by_id in family_samples_by_id.items():
+            for family_guid, samples_by_id in sorted(family_samples_by_id.items()):
                 quality_q = Q()
-                for sample_id in samples_by_id.keys():
-                    for field, config in quality_field_configs.items():
+                for sample_id in sorted(samples_by_id.keys()):
+                    for field, config in sorted(quality_field_configs.items()):
```

Comment:
The `sorted()` added here is required because there are `&` operations on query filters such as `quality_q &= ~Q(q) | ~Q('term', samples_num_alt_1=sample_id)` and `quality_q &= ~Q(q)`. 
I did an experiment which shows that query filter `&` operation is order dependent:
```
>>>sample1_q = Bool(should=[Term(samples_num_alt_1=u'HG00732'), Term(samples_num_alt_2=u'HG00732')])
>>>sample2_q = Bool(should=[Term(samples_num_alt_1=u'HG00731'), Term(samples_num_alt_2=u'HG00731')])
>>>sample2_q & sample1_q
Bool(
  minimum_should_match=1,
  must=[Bool(minimum_should_match=1, should=[Term(samples_num_alt_1=u'HG00732'), Term(samples_num_alt_2=u'HG00732')])],
  should=[Term(samples_num_alt_1=u'HG00731'), Term(samples_num_alt_2=u'HG00731')])
>>>sample1_q & sample2_q
Bool(
  minimum_should_match=1,
  must=[Bool(minimum_should_match=1, should=[Term(samples_num_alt_1=u'HG00731'), Term(samples_num_alt_2=u'HG00731')])],
  should=[Term(samples_num_alt_1=u'HG00732'), Term(samples_num_alt_2=u'HG00732')])
```
From Elasticsearch point of view, the results of `sample2_q & sample1_q` and `sample1_q & sample2_q` are the same. But, when making comparison for testing, they are different. To make the results predictable, we have to keep the order when use the `&` operator on query filters.